### PR TITLE
bugfix/19851-rangeselector-hover-state

### LINKED ISSUE #19851: RangeSelector button hover state gets destroyed when adding data to a series
#### Expected behaviour
The hover state of the RangeSelector buttons should not be influenced if data is added to a series in the background.

#### Actual behaviour
The hover state is not visible if the cursor hovers over a RangeSelector button while  data is added to a series.

#### Live demo with steps to reproduce
[](https://jsfiddle.net/stormtroopa/9v6n8z35/2/)

#### Product version
11.1.0

#### Affected browser(s)
Chrome

### PR DESCRIPTION
**Pull request by pawellysy**
Fixed #19851, the `rangeSelector` button state was not preserved after adding data to a series.

### CODE REVIEW

**Review by kamil-musialowski — APPROVED** [review]
Cool!

File: samples/unit-tests/rangeselector/update/demo.js [hunk 1]
```diff
@@ -59,3 +60,22 @@ QUnit.test('RangeSelector update', function (assert) {
...
+QUnit.test('RangeSelector update hover', function (assert) {
+    var chart = Highcharts.stockChart('container', {
+        series: [{
+            pointInterval: 24 * 36e5,
+            pointStart: Date.UTC(2013, 0, 1),
+            data: Array.from({ length: 100 }, (_, x) => Math.sin(x / 10) * 10)
+        }]
+    });
+    const controller = new TestController(chart);
+    controller.mouseEnter([100, 15], [100, 25], {});
```
[the remark this answers is not in the document]
Comment (pawellysy) [reply samples/unit-tests/rangeselector/update/demo.js:73]: Actually, the test does not work without this one, so this is an intentional change :D

[the remark this answers is not in the document]
Comment (kamil-musialowski) [reply samples/unit-tests/rangeselector/update/demo.js:73]: Oh, okay! Interesting :D

**Review by DJTechnoo — APPROVED** [review]
Looks good, approved! Just some comments though

**Comment by pawellysy** [comment]
@TorsteinHonsi @DJTechnoo The parameter is initialized without the value, so on the first render it's undefined, which causes the condition to always fail, and render the buttons, so it was desired choice.

**Review by TorsteinHonsi — APPROVED** [review]
@pawellysy Oh, sorry, I didn't see that the second was a strict check for `false`. Then I agree with you.

**Comment by TorsteinHonsi** [comment]
Now pending checks...

**Pull request merged**